### ISSUE TITLE
tags_find / tag_files broken with Nextcloud 25.0.1

### ISSUE THREAD

**Issue by EnderArchery** [Raudius/files_scripts#23]
I was working my way up to creating a script that would allow users to tag files for ocr-scanning.
And it was working.
But at some point, I could create a new public tag called ocr in the admin-panel, but
```
tags = tags_find({name = "ocr"})
log(#tags)
```
started to return 0.

And where I was getting a single tag before (tags_find({id = 5}))
became Nil (The id is identical to the one in the headers of the network traffic when modifying it)

**Comment by EnderArchery** [comment]
How to reproduce:
Create a public tag called ocr
create an action with the script
```lua
log("empty")
log(#tags_find())
log("visible")
log(#tags_find({user_visible = true}))
log("by name")
log(#tags_find({name = "ocr"}))
```
![image](https://user-images.githubusercontent.com/43007119/200091477-3fb6e871-dbec-42de-a161-0db5e71b941d.png)
![image](https://user-images.githubusercontent.com/43007119/200091508-408ecb4e-4135-495b-a0a2-ae0551fbefee.png)

**Comment by Raudius** [comment]
Hi, thanks for the report.

I think the issue here is that `tags_find` returns a table with IDs as keys. Lua's `#` does not calculate the length of these (non-sequential) tables.

```lua
{
  3= {
    id= 3,
    user_visible= true,
    name= "ocr",
    _type= "tag",
    user_assignable= true,
    access_level= 0
  }
}
```


You can instead check the result of your script by logging the JSON of the results

```lua
log("empty")
log( json(tags_find()) )
log("visible")
log( json(tags_find({user_visible = true})) )
log("by name")
log( json(tags_find({name = "ocr"})) )
```

**Comment by Raudius** [comment]
That said, I do consider this a bug, as there is no reason to not return a sequential array here. So I will change it for the next release :)

**Comment by EnderArchery** [comment]
You were right (sorry, I'm new to lua)

tag_files however seems to be broken then.
Same setup as before
```LUA
pdfMime = "application/pdf"
ocrTag = tags_find({id = 9})
files = get_input_files()

log(json({ocrTag = ocrTag}))

function scanFiles(files, layer)
  for fileIndex = 1, #files do
    local file = files[fileIndex]
    local fileType = meta_data(file).mimetype
    if(fileType == pdfMime) then
      log(json({file = file}))
      log(json({tag_file = tag_file(file, ocrTag)}))
    elseif(fileType == "httpd/unix-directory") then
      local folderFiles = directory_listing(file)
      scanFiles(folderFiles, layer + 1)
    end
  end
end

scanFiles(files, 1)
```

Then running the action on a single pdf file results in
tag_files returning false, although the tag is not assigned to the file yet and the user is able to see and assign the tag to the same exact file easily.

![image](https://user-images.githubusercontent.com/43007119/201404114-8d3c15e4-2b6f-4f3d-8e30-982fcb627eea.png)

![image](https://user-images.githubusercontent.com/43007119/201404252-5ade6156-7a4e-4dc5-99a6-16135ad70105.png)

![image](https://user-images.githubusercontent.com/43007119/201403971-2c14e86b-b279-4bc1-bdc8-e9301b3ace96.png)

**Comment by Raudius** [comment]
`tags_find` always returns an array of tags (even if only 1 tag is expected/returned). You can see in the log that `ocrTag` is a table which contains the tag table 
```lua
ocrTag = tags_find({id = 9}) 
log(json(ocrTag)) -- { 9= { ... tag details ... }|
```

You should try to extract the actual tag table. This way would be future proof for after I fix the `tags_find` behaviour:
```lua
ocrTags = tags_find({id = 9})
id, ocrTag = next(t)
```

**Comment by Raudius** [comment]
The fix for the `tags_find` issue is included in the latest release (2.0.0), so I will close the issue

Feel free to reach out if you're still having trouble with your script

**Comment by EnderArchery** [comment]
I can confirm it working on my end. Though 
I just changed it to
```LUA
ocrTag = tagsFind({id = 9})[1]
```
